src/stack_limiter/mod.rs

```diff
@@ -61,6 +34,43 @@ impl Context {
 	}
 }
 
+/// Represents the injection configuration. See [`inject_with_config`] for more details.
+pub struct InjectionConfig<'a, I, F>
+where
+	I: IntoIterator<Item = Instruction>,
+	I::IntoIter: ExactSizeIterator + Clone,
+	F: Fn(&FunctionType) -> I,
+{
+	pub stack_limit: u32,
+	pub injection_fn: F,
+	pub stack_height_export_name: Option<&'a str>,
+}
+
+/// Same as the [`inject`] function, but allows to configure exit instructions when the stack limit
+/// is reached and the export name of the stack height global.
+pub fn inject_with_config<I: IntoIterator<Item = Instruction>>(
```

Comment:
could you move it under `inject` function?

Reply:
done